common/djangoapps/student/views.py

```diff
@@ -469,9 +469,9 @@ def complete_course_mode_info(course_id, enrollment, modes=None):
         modes = CourseMode.modes_for_course_dict(course_id)
 
     mode_info = {'show_upsell': False, 'days_for_upsell': None}
-    # we want to know if the user is already verified and if verified is an
-    # option
-    if 'verified' in modes and enrollment.mode != 'verified':
+    # we want to know if the user is already enrolled as verified or credit and
+    # if verified is an option.
+    if 'verified' in modes and enrollment.mode not in ['verified', 'credit']:
```

Comment:
It seems safer and more intuitive to me if we expressed this like:
```
if 'verified' in modes and enrollment.mode in CourseMode.UPSELL_TO_VERIFIED_MODES:
```
where
```
UPSELL_TO_VERIFIED_MODES = ["honor"]
```

This has two advantages:
1) It's more explicit about which modes are allowed to upsell to verified, which protects against this same issue happening for other modes we may introduce.
2) It eliminates one more place in which we're hard-coding course mode keys.